app/Http/Controllers/Front/BaseCartController.php

```diff
@@ -212,22 +215,35 @@ public function updateAgentQty(Request $request)
     {
         try {
             $id = $request->input('productid');
-            $cartValues = $this->getCartValues($id);
-            Cart::update($id, [
-                'price'      => $cartValues['price'],
-                'attributes' => ['agents' =>  $cartValues['agtqty'], 'currency'=> ['currency'=>$cartValues['currency'], 'symbol'=>$cartValues['symbol']]],
-            ]);
+            $hasPermissionToModifyAgent = Product::find($id)->can_modify_agent;
+            if ($hasPermissionToModifyAgent) {
+                $cartValues = $this->getCartValues($id);
+                Cart::update($id, [
+                    'price'      => $cartValues['price'],
+                    'attributes' => ['agents' =>  $cartValues['agtqty'], 'currency'=> ['currency'=>$cartValues['currency'], 'symbol'=>$cartValues['symbol']]],
+                ]);
+            }
 
             return successResponse('Cart updated successfully');
         } catch (\Exception $ex) {
             return errorResponse($ex->getMessage());
         }
     }
 
+    /**
+     * The method returns the updated price, no of agents and currency of the product added to the cart
+     * Since this method is called when the the user has permission to modify agents(set from the admin panel), if the parameter $canReduceAgent is true the agent quantity and the cart total will be divided by two else they will be multiplied by two.
```

Comment:
reason for multiplication or divide by 2?

Reply:
updated